### PR TITLE
chore: improve naming and comments

### PR DESCRIPTION
**Pull request by sleepdefic1t**
## Summary

- use more descriptive naming.
- add comments.
- move APDU flag constants to an enum.
- increase APDU Payload max from 4 to 10 chunks.
- add more argument checks.
- use linting.

## Checklist

- [x] Ready to be merged

## Additional Comments

Changes are non-breaking.

### CODE REVIEW

File: src/ark.ts [hunk 1]
```diff
@@ -1,107 +1,247 @@
...
-			if (index === "0") {
-				writePathsToBuffer(paths, buffer);
+		/** build the Apdu payload chunks */
+		const bip32Path = parseBip32Path(path);
+		const chunks = splitApduChunks(
+			hex,
+			this.CHUNK_SIZE * 2 - (2 + bip32Path.length * 4) * 2
+		);
+
+		chunks.forEach(function (chunk, index) {
```
Comment (alexbarnsley) [inline src/ark.ts:193]: We tend to stick to for/in or for/of loops as those are generally faster than forEach

File: src/utils.ts [hunk 2]
```diff
@@ -1,34 +1,75 @@
...
+	elements: number[],
+	buffer: Buffer
+): void {
+	if (elements.length === 0 || elements.length > 10) {
+		throw new Error("Invalid Path Length");
 	}
+
+	buffer[0] = elements.length;
+
+	elements.forEach(function (element, index) {
```
Comment (alexbarnsley) [inline src/utils.ts:49]: We tend to stick to for/in or for/of loops as those are generally faster than forEach

File: src/ark.ts [hunk 3]
```diff
@@ -1,107 +1,247 @@
...
-	readonly PAYLOAD_MAX = 255 * 4;
 	private transport: LedgerTransport<any>;
 
-	constructor (transport: LedgerTransport<any>) {
+	readonly CHUNK_MAX: number = 10;
+	readonly CHUNK_SIZE: number = 255;
+	readonly PAYLOAD_MAX: number = this.CHUNK_MAX * this.CHUNK_SIZE;
+
+	/**
+	 * ARK Ledger Transport Class Ctor.
```
Comment (alexbarnsley) [inline src/ark.ts:90]: I don't think this is necessary

**Comment by alexbarnsley** [comment]
Can't you make branches? The fork isn't necessary 🤔

File: src/ark.ts (hunk 3 above)
Comment (sleepdefic1t) [reply src/ark.ts]: agreed, this one slipped though. C terms don't belong here.

**Comment by sleepdefic1t** [comment]
@alexbarnsley I made a fork is so I could use Codacy/CodeFactor/CodeClimate/etc.

requested changes have been pushed, and is tested as working 👍

**Comment by alexbarnsley** [comment]
All looks okay and loading the ledger in the desktop wallet now. I've sent @samharperpittam the build for testing so won't approve until he's happy 👌

**Pull request merged**